client/sites-dashboard/components/searchable-sites-table.tsx

```diff
@@ -24,7 +28,17 @@ export function SearchableSitesTable( { sites }: SearchableSitesTableProps ) {
 		return searchCollection( sites, term.toLowerCase(), [ 'URL', 'domain', 'name', 'slug' ] );
 	}, [ term, sites ] );
 
-	const handleSearch = ( rawTerm: string ) => setTerm( rawTerm.trim() );
+	const handleSearch = ( rawTerm: string ) => {
+		setTerm( rawTerm.trim() );
+		const encodedTerm = encodeURIComponent( rawTerm.trim() );
+		if ( encodedTerm.length ) {
+			page(
```

Comment:
Calling `page()` this frequently is also causing a lot of pretty big renders. If you take a look at the rendering in the React dev tools you can see _everything_ on the page is rerendering when the `page()` function gets called. Which makes sense I guess because it's going right back up to the controller level.

(I'm seeing what gets re-rendered by using this setting in the dev tools)
![Jul-18-2022 21-08-50](https://user-images.githubusercontent.com/1500769/179480720-db237565-f4c4-49d4-83b7-008e123261d1.gif)

Before this PR, it's only the search box and table that re-renders as the user types. Obviously it'd be good to re-render as little as possible as the user types so that it feels as responsive as possible. I'm not sure if maybe we can maybe bypass the controller code by calling `history.pushState()` directly? Or maybe we could only call it after a timeout?

Reply:
> The dashboard components would be more easily embeddable in other contexts if they didn't use `window.location` and `page()` directly but instead called up to their parent. But we can also figure that out when we refactor the components back into `@automattic/components` again.

Good point. I've pushed this discussion to its own issue #65674

> (I'm seeing what gets re-rendered by using this setting in the dev tools)

Nifty! Do you leave that on all the time, or only when you see `page()` being called?

> Before this PR, it's only the search box and table that re-renders as the user types. Obviously it'd be good to re-render as little as possible as the user types so that it feels as responsive as possible. I'm not sure if maybe we can maybe bypass the controller code by calling `history.pushState()` directly? Or maybe we could only call it after a timeout?

`history.pushState()` seems to work fine: a666dc12a5

Reply:
> Do you leave that on all the time, or only when you see `page()` being called?

No way, too much flashing! 😄
I take a look whenever I suspect an expensive update is happening. In this case I wasn't call to `page()` that tipped me off to investigate the size of the re-render. It was the fact we're updating things on keypress.